msticpy/data/core/data_providers.py

```diff
@@ -193,12 +196,13 @@ def add_connection(
 
         """
         # create a new instance of the driver class
-        new_driver = self.driver_class(**(self._driver_kwargs))
-        # connect
+        new_driver = QueryProvider(self.environment,
```

Comment:
This was changed since the previous method to initialise he driver resulted in the incorrect dataprovider being used. Ie. when initialising `QueryProvider("M365D")` every additional connection used a different API endpoint for the queries (securitycenter.microsoft.com) since `self.data_environment` was set to none (for the additional connections).

Reply:
I think that makes sense. Definitely.